packit_service/worker/whitelist.py

```diff
@@ -0,0 +1,202 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import enum
+
+import requests
+import logging
+
+from frambo.dict_in_redis import PersistentDict
+
+logger = logging.getLogger(__name__)
+
+
+class WhitelistStatus(enum.Enum):
+    approved_automatically = "approved_automatically"
+    waiting = "waiting"
+    approved_manually = "approved_manually"
+
+
+class GithubAppData:
+    def __init__(
+        self,
+        installation_id: int,
+        account_login: str,
+        account_id: int,
+        account_url: str,
+        account_type: str,
+        created_at: int,
+        sender_id: int,
+        sender_login: str,
+        status: WhitelistStatus = WhitelistStatus.waiting,
+    ):
+        self.installation_id = installation_id
+        self.account_login = account_login
+        self.account_id = account_id
+        self.account_url = account_url
+        self.account_type = account_type
+        self.created_at = created_at
+        self.sender_id = sender_id
+        self.sender_login = sender_login
+        self.status = status
+
+    def get_dict(self) -> dict:
+        result = self.__dict__
+        # whole dict have to be JSON serializable because of redis
+        result["status"] = str(result["status"])
+        return result
+
+
+class Whitelist:
+    def __init__(self):
+        self.db = PersistentDict(hash_name="whitelist")
+
+    @staticmethod
+    def _is_packager(account_login: str) -> bool:
+        """
+        If GitHub username is same as FAS username this method checks if user is packager.
+        User is considered to be packager when he/she has the badge:
+         `If you build it... (Koji Success I)`
+        :param account_login: str, Github username
+        :return: bool
+        """
+
+        url = f"https://badges.fedoraproject.org/user/{account_login}/json"
+        data = requests.get(url)
+        if not data:
+            return False
+        assertions = data.json().get("assertions")
+        if not assertions:
+            return False
+        for item in data.json().get("assertions"):
```

Comment:
There could be `assertions` and not calling `data.json().get("assertions")` again.